### ISSUE TITLE
Rethinking 3.6's Dashboard Ads (#22857 )

### ISSUE THREAD

**Issue by justlevine** [woocommerce/woocommerce#23198]
#22857 seemed to slip through under the radar, and I really hope it's not too late to reconsider this.

1. If it's using the WC API to display these ads, what's the impact regarding GDPR?

2. Am I blind, or is there no way to dismiss this from Products/Orders dashboard? 

Ignoring that this entire PR seems to run contrary to the WP.org guidelines ( https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#11-plugins-should-not-hijack-the-admin-dashboard ) there's really no user benefit to having them on those screens.

You could argue that on the Create/Edit screens an ad might make sense (since the user might be looking for functionality that doesn't come baked in), but if someone is on the Product/Order list screens they're looking to manage their site, and anything that gets in the way of that workflow (let alone something non-dismissable) is the definition of intrusive.

~~3. Unless I missed it, theres no filter to disable these ads.~~ (`woocommerce_allow_marketplace_suggestions` is seemingly the filter that lets you disable them. Thanks @hedqvist for the find!)

4. It's one thing for Jetpack to start hijacking search results with anti-competitive ads. People like to sh*t on Jetpack anyway, so what difference does it make. But surely in the wake of that backlash, ya'll can see how alienating something like this is. Automattic is under enough flak from the community for the percieced hijacking the open-source goals of WP to make a quick buck (*cough* Gutenberg's half-baked release who's success was predicated on biweekly iteration *cough*). And surely there's enough ux research that shows these sorts of heavy-handed advertisements turn users away. How can anyone think this is a good idea?

**Comment by jkohlbach** [comment]
The whole thing is against WordPress plugin guidelines. It clearly says no hijacking in the policy. If WooCommerce gets a free pass on it, should other plugins?

In addition, didn't anyone think it might be a conflict of interest for WooCommerce the commercial entity to use WooCommerce the open source plugin to show ads in this manner? Bit anti-competitive to all the 3rd party devs out there (of which there are a lot).

WooCommerce already has an amazing page under WooCommerce->Extensions with full searching capabilities etc. Why would you want to show **irrelevant** ads during a user's everyday workflow?! Store owners use these screen daily, it's **terrible** UX.

I suggest that it gets ripped out in its entirety and filed under "cool implementation/fun to code but horrible idea for actual users"

**Comment by tonybahama** [comment]
Oh god please remove the ads... they don’t make sense anyways since our emoloyees cannot decide what plugins we use.
Annoying as hell!:(

**Comment by jonathan-dejong** [comment]
I'm genuinely mad at this PR making it through under the radar like this!
There is one thing we constantly joke about in a bitter way at my agency and that is the incessant promotional _crap_ a lot of plugin merchants try to inject in our admin interface. We've even developed a plugin called notices trashcan to somewhat handle it for our clients..

I implore you to dismiss this horrible mistake of an PR and instead focus on the already existing extensions subpage (which we also never use because we KNOW what we want/need for each site we build).

Or at the very least you could provide a filter to completely disable these ads as per the Ops request. So we professionals who build and maintain sites for our clients can get rid of these intrusions from the work flow.

**Comment by ErikBernskiold** [comment]
I too am very concerned about this. I get that WooCommerce want to benefit from their commercial side too, and there are many ways to do this.

But in this case, it feels like this is at a great disregard for the users. Hijacking a product list, order list or a user interface element in this way is a major interruption of the user experience. It's not the place for an ad.

There might be legitimate reasons to argue for hijacking a list like this, but for an ad (that's also sending an external requests)... I can't think of one good reason where this can be considered "helpful UX".

The only reason for sneaking a feature in, is when you know it will be disliked.

I would expect an Automattic product to lead the way in inplementing the WordPress guidelines in the handbook. Not using the Automattic position to do, and I hate to be a little harsh, do whatever you like just because.

If there is a good justification for why this is a good user experience where these ads are placed at the moment, I'm all ears and happy to reconsider my position.

**Comment by nicomollet** [comment]
I am really concerned about this too.
Products and Orders should not suffer from addon suggestions or paid options advertizing at all.

**Comment by hedqvist** [comment]
I'm also not very happy about this change, because users shouldnt have to see this and interuppt this as they are doing their daily work.

When going through the initial PR ( #22857 ) there seems to be a filter that can be used to block this out. https://github.com/woocommerce/woocommerce/blob/master/includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php#L161-L183

```woocommerce_allow_marketplace_suggestions```
Perhaps that could be of use @jonathan-dejong @ErikBernskiold

**Comment by pmgarman** [comment]
I do not support WooCommerce plastering the admin with ads for it's own extensions. We took a not terrible but not great admin interface, and instead of focusing on improving it, we stuck ads all over it.

**Comment by JPry** [comment]
There is only one scenario where I think this feature should stay in place and could be beneficial: If this feature is controlled by an **opt-in** toggle in WC settings. Otherwise, this whole feature should be removed entirely. Store owners don't need advertisements in their admin dashboard. In my opinion, leaving this feature in place would be very harmful to the WooCommerce community.

**Comment by adampickering** [comment]
I don’t think it technically violates the guidelines it’s just obnoxious and makes WooCommerce look like a low rent solution. It seems we are in a midst of a monetization push and they are looking for any where they can add up sells. Apparently doing so gracefully has gone out the window.

And if you are phoning home to your API without first asking the user for permission to do that then it will likely be removed from the directory after release.

**Comment by phpbits** [comment]
It's a horrible idea, please don't start this admin hijack. If this promotion continue I'm afraid more plugins will do the same and it's not beneficial for end users.

**Comment by Ipstenu** [comment]
​It appears what you're doing here is tracking usage, reporting that back to your API, and returning the relevant features. And the problem there is you're running right up into tracking users **WITHOUT** consent. Welcome to GDPR hell.
​
​Ads are a VERY touchy subject for a lot of reasons, but the primary one here is legal. That is, if you're transmitting data about my actions without letting me opt IN to this, or without it being a required aspect of the plugin (and no, it's not) then you're violating my privacy.

Also yes, these need to be dismissible or 'solvable' - If an alert is there because there's a problem (like "You must enable pretty permalinks...") then it's fine not to have it dismissible. The alert will self-resolve once you enable permalinks. On the other hand, an 'optional notice' like "Try this feature!" should have a "no thanks" dismiss to not clutter up the dashboard.

As always, Woo, you can email `plugins@wordpress.org` before you implement this kind of thing to ask if it's okay. Our normal recommendation is pretty simple:

1) Don't track people without consent
2) Notices have to be dismissible or resolvable
3) Don't annoy your users, they leave you bad reviews and we won't delete them

We're not gonna bail anyone out of their bad PR run.

**Comment by timmyc** [comment]
:wave: hello all, 3.6 Release Lead here to chime in and say we are most definitely listening to the feedback on this feature, and value everyone's input. @Ipstenu thanks for reiterating the plugin guidelines too.

I wanted to hop into the conversation here and provide some tech details on what has been implemented and slated for 3.6 - so everyone is equipped with all the details - because as we all know 

> Ads are a VERY touchy subject for a lot of reasons, but the primary one here is legal.

First off I wanted to highlight the fact the development team that worked on this feature did reach out to the plugins team prior to commencing development. And the plugins review team replied with the same guidelines that Mika [shared above](https://github.com/woocommerce/woocommerce/issues/23198#issuecomment-479536680).

Now to an overview of what is happening behind the scenes of Marketplace Suggestions:

- Suggestions are only enabled by default for a subset of users. The full logic behind this can be [seen here](https://github.com/woocommerce/woocommerce/blob/5492369fe935b94aa6a06e1acbd2a58ea0fc09c3/includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php#L161) But tl;dr; the feature is only enabled for English locales, and suggestions are only shown to users whom have the capability to install plugins.
- Additionally, suggestions can be disabled via the [`woocommerce_allow_marketplace_suggestions` filter](https://github.com/woocommerce/woocommerce/blob/5492369fe935b94aa6a06e1acbd2a58ea0fc09c3/includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php#L182)
- Suggestions are fetched from a `.json` file that is hosted on woocommerce.com. The code that handles that request can be seen in [this method](https://github.com/woocommerce/woocommerce/blob/5492369fe935b94aa6a06e1acbd2a58ea0fc09c3/includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php#L41). Please note that **no data is transmitted in this request about the site, nor the user**.
- That data is then used to determine if suggestions should be shown for a given context [in this javascript logic](https://github.com/woocommerce/woocommerce/blob/5492369fe935b94aa6a06e1acbd2a58ea0fc09c3/assets/js/admin/marketplace-suggestions.js#L247). This code does take into consideration currently installed extensions to determine if suggestions should be shown.
- All suggestions are dismiss-able per the plugin repository guidelines.
- When a suggestion has been dismissed for a given area ( i.e. Product Listing page ), no further suggestions are shown for 24 hours. If it is dismissed five times, none are shown for a month. This code can be [seen here](https://github.com/woocommerce/woocommerce/blob/master/assets/js/admin/marketplace-suggestions.js#L407)

This feature does record Usage Tracking, but I also wanted to detail a bit more about that side of the discussion to ensure complete clarity on how it works too. An email was just sent to all WooCommerce customers with some information about this too, and I suggest [everyone reads our documentation page](https://woocommerce.com/usage-tracking/) about it as well.

- Usage tracking in WooCommerce is only performed if a site is opted-in.
- The opt-in dialog is shown in the Setup Wizard ( [code here](https://github.com/woocommerce/woocommerce/blob/ae1841a82cf6736fdc8a3fc61e0f51bcfadaf9d0/includes/admin/class-wc-admin-setup-wizard.php#L525-L536) )
- In enhancement in 3.6 is making it easier for sites to have full control over opting in/out, and a setting has been added to the _Settings | Accounts & Privacy_ tab:

![image](https://user-images.githubusercontent.com/22080/55494913-9a416600-55f0-11e9-968f-c84fb5fdb9f6.png)

I hope this provides some more technical information around what has been implemented. The team is listening to all of the feedback here ( and in various other venues ), and value it very much.

**Comment by DavidAnderson684** [comment]
> "If you would rather opt-out, and do not check this box, we will not know this store exists and we will not collect any usage data."

Is this entirely true, given that by default, WP sends a list of active plugins (e.g. WooCommerce) and the number of users on a site (i.e. generally the number of customers) to wordpress.org daily? Does the the statement rely on a subtle understanding of who is meant by "we" ? (i.e. "we" is Automattic, but wordpress.org is technically a separate legal entity, though a number of Automattic employees work there and presumably (?) have access to the data)?

**Comment by DavidAnderson684** [comment]
@timmyc What this all seems to boil down to is that, for store owners who don't know about filters (or searching for plugins to provide solutions), after an initial burst of dismissing notices daily, they're then either
a) going to be dismissing an advert every 6 days or
b) going to receive intensive/continual training in ignoring dashboard notices on WooCommerce pages.
Both of those outcomes seem terrible.

**Comment by GlennMartin1** [comment]
In the email, it says:

> If the above removal-by-script option proves to be difficult to implement – for example, for those who are not comfortable adding custom code – we will explore introducing a simpler way to turn them off and include this in a point release (e.g. a toggle in core settings).

Yes, I would like to see this become a toggle in core before 3.6 is released.

**Comment by pmgarman** [comment]
I think we need to re-evaluate the definition of dismissable.

> When a suggestion has been dismissed for a given area ( i.e. Product Listing page ), no further suggestions are shown for 24 hours. If it is dismissed five times, none are shown for a month.

Dismissing just to keep hounding the user, that's not dismissing... that is snoozing. Because I told you 5 times that I don't want to see your ads, that doesn't mean come back in a month. The average user shouldn't have to use a filter to make ads go away.

**Comment by JPry** [comment]
@timmyc Thank you for the feedback. I think that one addendum needs to be added to your list above about site tracking. 

> * Usage tracking in WooCommerce is only performed if a site is opted-in.
> * The opt-in dialog is shown in the Setup Wizard ([code here](https://github.com/woocommerce/woocommerce/blob/ae1841a82cf6736fdc8a3fc61e0f51bcfadaf9d0/includes/admin/class-wc-admin-setup-wizard.php#L525-L536))
> * In enhancement in 3.6 is making it easier for sites to have full control over opting in/out, and a setting has been added to the Settings | Accounts & Privacy tab:

It should also be noted that if you go to WooCommerce > Extensions > WooCommerce.com Subscriptions, and click the "Connect" button, tracking will automatically be turned on **without opt-in**, as seen here:

https://github.com/woocommerce/woocommerce/blob/c7fc4a9521f56fd3dc272d96918a93544231d7b6/includes/admin/helper/class-wc-helper.php#L744-L748

**Comment by adampickering** [comment]
Can we change the wording to "Snooze" instead of dismiss? Cos I agree with @pmgarman when I hide something and then it comes back i'm going to get SUPER annoyed, then having to use a function to hide it forever seems very unfriendly to a user who may not be that technical.

**Comment by justlevine** [comment]
First off thanks so much @timmyc for taking the time to respond and clarify things! Especially the assurances regarding the lack of PID being sent to the WC servers...

> * All suggestions are dismiss-able per the plugin repository guidelines.

Im happy to hear you say this. In the RC-1 on my localhost server (and confirmed in a poopy.life install I just spun up), I see no way to dismiss these ads from either the Products or the Order Pages (only on the create/edit screen) [screenshot](https://imgur.com/a/IqqRP4G). I guess Ill open up a separate bug report.

> * When a suggestion has been dismissed for a given area ( i.e. Product Listing page ), no further suggestions are shown for 24 hours. If it is dismissed five times, none are shown for a month. 

It seems to me from the code, that after a month `contextDismissalCountCookie` resets back to 0, meaning you'd have to repeat the 5 consecutive snoozes before another month of quiet. Isn't this a bit heavy-handed? (Also agree with @pmgarman, and ps if ya'll are insistent on going forward with this, perhaps a 'Dismiss Permanently' option could be added...).

[From the guidelines
](https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#11-plugins-should-not-hijack-the-admin-dashboard)
> Advertising within the WordPress dashboard should be avoided, as it is generally ineffective. Users normally only visit settings pages when they’re trying to solve a problem. 

Have ya'll considered or done some user research on the effects of these sorts of ad placements? Putting aside whether these ads will even be effective, what are the implications of interrupting an already pretty complex workflow?

I can (partially) understand showing these sorts of ads in the Create/Edit screen, since they can theoretically prompt a solution to what the user is trying to do (even if it disrupts their workflow by taking them offsite > purchase page > plugin installation/activation > and then back to the Create/Edit flow).  But under what circumstance would ads like this be anything other than detrimental to the user's workflow the Products/Orders screens (where we go for for the daily management of our store, not its the config/development )?

If you wanted to upsell in a way thats __actually__ beneficial to the user, the place for these ads would be where they are offering an immediate and relevant solution, I.E. the various setting/configuration screens. Advertise payment gateways on the Settings\Payment screen, Shipping plugins on the Shipping screen, Marketing on the email screen, and if you must Product extensions on the Create/Edit Product screen (unlike the random list there now that doesn't necessarily have anything to do with product configuration). They're not doing anyone any good how they are now.

Again, thanks for the transparency. There's nothing necessarily wrong with ya'll trying to squeeze out some more money from users - **so long as its done tastefully, and in a way that actually provides value to the user,** instead of spamming and hindering them.

**Comment by timmyc** [comment]
> In the RC-1 on my localhost server (and confirmed in a poopy.life install I just spun up), I see no way to dismiss these ads from either the Products or the Order Pages (only on the create/edit screen) screenshot. I guess Ill open up a separate bug report.

Please do open up a bug report on that - and remember to include your browser details as I haven't seen it look like that in my testing of the feature.

**Comment by Ipstenu** [comment]
> When a suggestion has been dismissed for a given area ( i.e. Product Listing page ), no further suggestions are shown for 24 hours. If it is dismissed five times, none are shown for a month. This code can be seen here

That's bumping up on the intent of not making a user's life miserable. Can you imagine if you got an email every day and only after 5 would it stop for a month and then start again? Yikes.

I understand the purpose here, but it's pushing right up on making the dashboard unenjoyable to use. When users don't like doing a thing, they don't do it. That means they may not only stop using your plugin but WordPress.

> Putting aside whether these ads will even be effective, what are the implications of interrupting an already pretty complex workflow?

The worst case scenario is people quit Woo and WP.

**Comment by lukecav** [comment]
Feels like a feature which should be both opt-in as well as something that can be permanently disabled. I get why it is being done, but not how it is being done.

**Comment by timmyc** [comment]
> It should also be noted that if you go to WooCommerce > Extensions > WooCommerce.com Subscriptions, and click the "Connect" button, tracking will automatically be turned on without opt-in, as seen here:

@JPry thanks for pointing that out - let me research the history of that a bit as it appears that particular line of code has been in place long before 3.6 ( #15519 )

**Comment by BurlesonBrad** [comment]
So, it's whack-a-mole with the drama club and the lawyers with the rabbit hole of the run-around with the disclaimer of the opt-in of the other rabbit hole from the drama club and the whack-a-mole department.

Give it up already. Go run for president or something.

add_filter('woocommerce_allow_marketplace_suggestions', '__return_false');

**Comment by Ipstenu** [comment]
> This feels on the up-and-up to me as clear consent is given when clicking Allow. 

De-facto consent, versus actual optional consent.

It's like the whole "Hey, we use cookies, click "I Accept" or go away!" thing. Is that really consent? It's forced to consent, and that's really distasteful.

**Comment by BurlesonBrad** [comment]
> De-facto consent, versus actual optional consent.

_(you're being too kind)_

This is a backtracked ad, justified by a maybe consent, along with a lose fitting justification, on top of an opt-in that's approximately 54,269,078 degrees of separation away, which is along-the-same path of **"Accept it or go away"**, complete with a dose of ill-thought-out planning, which will _(probably)_ be rationalized with a lame excuse, .......and more.

It's like the Classic Editor plugin which is about to overtake the #-of-downloads for JetPack, Akismet, and Yoast - Someone at the top is gonna say, "Meh........ Shove it down their throats.  This is a great idea.  Let's **DO** this!"

On top of stopping the excessive tracking, now we just need to add this as well:

`add_filter('woocommerce_allow_marketplace_suggestions', '__return_false');`

**Comment by Ipstenu** [comment]
It costs me very little to be polite and assume the best in humans. Everyone should try it for a change :)

**Comment by BurlesonBrad** [comment]
If _"assume the best"_ came within a Texas mile of some folks cognitive radar, this thread would **NOT** exist in the first place 😉

**Comment by agusmu** [comment]
new post from WooCommerce

https://woocommerce.wordpress.com/2019/04/03/extension-suggestions-in-3-6/

**Comment by BurlesonBrad** [comment]
> i think I will use the filter to hide marketplace suggestion to all of my Premium WooCommerce Themes...

Right. However, _(back to that whole "assume the best" concept)_ for the store owner who is running a business, who doesn't even want to wonder what that is, who doesn't tinker w/ code, who doesn't understand that concept of `filters` .... from their point of view:

Imagine owning a business: Opening up your store every day and walking into the warehouse to see a new sign, a different table, a weird display, something moved, & a shelf out of place, but no **reasonable** way for the 👉store owner👈 to make the abnormality stop.

I'm all for them promoting their business, and @agusmu just b/c you make your own extensions isn't really the 'best' objection. The bigger picture here is that it shouldn't be inside the store period.

Write an email, retarget on Facbook, have affiliates, send a pigeon, send a fax, write it on a post it pad, send a signal up in smoke, tap it out in Morse code,  whatever else.

But to just jab the ad in there? There's no justification where this is "the best" at all
![image](https://user-images.githubusercontent.com/6208757/55520813-90951e00-5643-11e9-999e-6a3239185373.png)

**Comment by justlevine** [comment]
[From the explainer post](https://woocommerce.wordpress.com/2019/04/03/extension-suggestions-in-3-6/):
> ...only on screens where we think the action will be relevant (for example, extensions to enhance product pages when going to edit products).

Im assuming [this json of recommendations](https://woocommerce.com/wp-json/wccom/marketplace-suggestions/1.0/suggestions.json) is still being tweaked? Because putting aside all the aforementioned workflow misgivings of these appearing anywhere other than the settings or empty-state Products/Orders screens (this one I actually think is a great idea, as it provides an actionable step for users to take that doesnt require disrupting the workflow): 

- Product Addons, Additional Variation Images, Min-Max-Quantities, Name Your Price,  shouldnt be on the products list, but only on the Create/Edit product screen 
- One Page Checkout isn't relevant to either the Products Lists or Product Meta Tab.

**Comment by agusmu** [comment]
> I'm all for them promoting their business, and @agusmu just b/c you make your own extensions isn't really the 'best' objection. The bigger picture here is that it shouldn't be inside the store period.
> 
> Write an email, retarget on Facbook, have affiliates, send a pigeon, send a fax, write it on a post it pad, send a signal up in smoke, tap it out in Morse code, whatever else.
> 
> But to just jab the ad in there? There's no justification where this is "the best" at all

to be honest, I don't know what to say after reading this article, https://woocommerce.wordpress.com/2019/04/03/extension-suggestions-in-3-6/ it also keep saying that this feature follows plugin guidelines...

i also join some FB group where I see some plugin developers start saying, "should we start do the same?"...

i know it is not "best" objection for me, but it is all what I can do for now...

let's see if WooCommerce would re-consider this feature...

**Comment by justlevine** [comment]
@timmyc  - 

Regarding **usage tracking**:
I'm unclear what exactly is collected regarding these advertisements if Usage Tracking is enabled.

Theres nothing mentioned regarding the information gathered or the events tracked when displaying, clicking on, or dismissing an advertisement in the [Usage Tracking documentation](https://woocommerce.com/usage-tracking/) you shared. Which seemingly runs counter to what you wrote in your first comment: 'this feature does record Usage Tracking'....

So what exactly is recorded, and when?

Also, regarding **consent**:
Is it safe to assume that since this (and the other new tracking events being added in 3.6) wasn't part of the initial consent that users who gave consent in a prior WC version would be required to reconsent before you begin collect this additional data?

**Comment by jkohlbach** [comment]
We're getting off track here.

The feature isn't compliant with WordPress's plugin development guidelines and is clearly hijacking an interface (regardless that the interface is WooCommerce's) with ads that are neither contextual or done "sparingly". It's honestly the most heavy handed approach to ads in a .org plugin that I've ever seen.

Ads like this do not help the store owner beyond the first impression and thus are irrelevant on the whole.

And re the ability to turn off... it's not currently opt-in, it's opt-out and as @BurlesonBrad summed up in this line: "store owner who is running a business, who doesn't even want to wonder what that is, who doesn't tinker w/ code, who doesn't understand that concept of filters" ... they're just gonna end up thinking WooCommerce is pretty shit for slamming these ads down their throat all day every day while they're trying to get their work done.

**Comment by franticpsyx** [comment]
Adding a few thoughts here.

A few releases ago, the **WooCommerce > Extensions** screen was added to WooCommerce, which gave store owners a tasteful, convenient way to discover functionality they might need:

<img width="1410" alt="Image 2019-04-04 at 9 22 20 a m" src="https://user-images.githubusercontent.com/1783726/55534180-48fea880-56bc-11e9-9203-e2c916064c2f.png">

This was generally nicely done, and it would have been great to see some effort being directed to improving the experience there:

* Make review scores (and maybe comments) visible.
* Add images/galleries to Extensions in collaboration with Extension developers.
* Maybe add some extra actions instead of just a link to the Extension page.
* Improve the search results by working with Extension developers to add tags/keywords to their products.
* Improve the design.
* Add some basic copy -- why should a store owner prefer the WooCommerce.com offering?

On top of these improvements, it would have been great to make an effort towards making the section more visible: 

For example, a tasteful notice could have been added to the Dashboard to direct store owners to this section. Having to snooze it a few times before dismissing it might have been a good idea, considering that most new store owners need weeks to understand their needs fully.

The additions discussed here follow a different path: The ads introduced in the Products table may manage to win the "fight for attention" by escaping the notices section at the top of the screen, but can be disruptive to existing store owners.

Might be a good idea to **ensure these messages are displayed in new installations only**. New store owners are very likely to find the information helpful. The **WooCommerce > Extensions** screen has been always very easy to ignore.

These messages will probably do more damage than good if they are displayed to store owners that have settled well into their workflows and routines.

Would love to see improvements to the **WooCommerce > Extensions** screen in the future! Happy to contribute, too.

**Comment by DavidAnderson684** [comment]
> Is it safe to assume that since this (and the other new tracking events being added in 3.6) wasn't part of the initial consent that users who gave consent in a prior WC version would be required to reconsent before you begin collect this additional data?

Under the EU's GDPR law, "assumed expanding consent" isn't allowed, even if it's somewhat similar. If some new data is being gathered that was not previously explicitly consented to, then new consent must be gained. Of course "somewhat similar" is vague. The specifics matter (and GDPR requires an explicit/formal evaluation process - they thought of the "oh, we didn't realise..." loophole and made it illegal).

**Comment by helgatheviking** [comment]
I find these ads to be _very_ intrusive, and that you need code to make them go away is annoying for all the folks who aren't comfortable with code. And having to repeatedly dismiss suggestions? That's intentionally antagonizing the users. 

If this is meant to help users find more extensions from the marketplace, maybe the effort would be better spent on the Extensions tab and/or in a better search/grouping/discovery process at the WooCommerce.com website itself.

**Comment by norcross** [comment]
putting aside the other (numerous) issues, the idea that we’re opting people in based on a button in a configuration wizard they (or someone else) may have clicked a year or two ago just seems _wrong_. the user didn’t agree to this feature. 

whether Woo believes they’ve already opted in or not in frankly irrelevant.

**Comment by BurlesonBrad** [comment]
> a year or two ago
### Exactly!
> 54,269,078 degrees of separation

Remind the user again what they did to opt into these bonus ads:
Was it consent on WordPress.com? _(whack a mole)_
Was it consent on WooCommerce onboarding wizard? _(whack a mole)_
Was it consent on WooCommerce.com prior-to their API based connection? _(whack a mole)_
Was it consent on JetPack.com 10+ years ago? _(whack a mole)_
Was it consent on Bubble Storm's site? _(whack a mole)_

While @norcross comment is a bit off from the original intent of **"Dashboard Ads"**, the point here is actually demonstrated quite eloquently.  

- Not only are the new tracking options added to 3.6.  _But wait, there's more...._
- `$on_top_of_that`, the location the store owners need to 'toggle' to switch off the monitoring is going to be relocated to a new tab & new location.  _But wait, there's **still** more..._
- ...... **AND** there's dashboard ads 😉 

Did you folks see this new book that came out?
![image](https://user-images.githubusercontent.com/6208757/55564064-b6aad480-56bc-11e9-828f-dcd51c22aaa4.png)

**Comment by DavidAnderson684** [comment]
N.B. If consent is being used as the basis for data processing, then GDPR requires the entity gathering the data to record and retain information on when and how consent was gained.

That inevitably leads to the follow-up question "what if it was gained a long time ago and we didn't keep records?". GDPR's answer is "then you absolutely may not process data upon that basis." i.e. Must treat it as if there was no consent, which means either deleting any such data or immediately requesting fresh consent.

**Comment by seb86** [comment]
![image](https://user-images.githubusercontent.com/6208757/55520813-90951e00-5643-11e9-999e-6a3239185373.png)

It makes me feel like I am not using WooCommerce fully but a sample of or trial of WooCommerce and all these extensions are required in order to use it. Does this appear for new users or all users? Honestly, just remove it. It's a mistake in marketing and not friendly to the shop owners especially if they are new to WooCommerce and WordPress and starting a new business. This would of driven me away if I was a newbie and taken my business elsewhere.

I agree with what @franticpsyx has explained. The extensions page should have been improved and he makes some great points to what could be done. At least there I don't feel like I am being misguided as to what I have in the core of WooCommerce before expanding my options.

I vote for a NO PUSH 👎

Let's re-think this. Take a step back. Put yourself in a different role. (Developers and Marketers excluded) 

Lets try this and see what results we get.

1. Bullet point changes you would want from a Shop Owners role. (For me it's the extensions page)
2. Bullet point changes you would want from a new WordPress user.
3. Bullet point changes you would want from an experienced WordPress user.
4. Bullet point changes you think would help the user based on time spent using WooCommerce.
5. Bullet point changes you think would help the user based on the amount of products they have active to manage.
6. Bullet point changes you think would help the user based on sales every quarter.

**Comment by BurlesonBrad** [comment]
..........or........ how about a wc version release just sticks to code.  that shouldn't be a foreign concept in the first place.  push out the code and move on: https://github.com/woocommerce/woocommerce/tree/3.6.0-beta.1

**Comment by seb86** [comment]
![tracking](https://woocommerce.com/wp-content/uploads/2019/03/WooCommerce-3.6-Usage-Tracking1.png)

The copy for this option does not read well. I'm dyslexic and this confuses me. I had to read it 3 times slowly just to understand it.

If this is staying. Change it to the following please. 

> Do not check this box if you do not wish to opt-in for WooCommerce.com to track your usage data. We will not know you store exists or provide you recommendations.

**Comment by itsgauravjain22** [comment]
I won't update woocommerce until this situation got fixed. The ads in platform is downgrading not upgrading the system.

**Comment by E-VANCE** [comment]
I get it – people invest money in Automattic and want to see some return and having bought WooCommerce for > 30 Mio. somebody is exploring new venues of income / marketing options and the official marketplace is presenting great potential.

But: This whole open-source thing / ecosystem resolves around trust and nagging, pushing, shoving, forcing won't do the trick, not in the short run and most certainly not in the long run. Don't put your brand, your community, your (previous) high standards on the line just to satisfy the quick buck.

It feels like some kind of law to me that once a company gets too big / owns a majority stake in some area and money is talking then they loose the sense of what made them create their product in the first place and focus on all the wrong stuff.

Sorry for 'ranting' but I am disappointed in how this is being handled.

**Comment by ajgw** [comment]
I was reading this whole thread and was between frustrated and angry what would be in 3.6. I see many well known devs who have well selling extension for wc at woocommerce.com and they argument against ads. Me too, but I am only a storeowner. And I had searched for extensions for features I needed at woocommerce.com and few times under woocommerce -> extensions screen. I use that more often. It is a good place, but yes it needs to have a better experience.

But think why do extensions to be promoted more?

My experience with extensions from woocommerce.com in the past were few times really frustrating. For example Follow-ups: I can not recommend it, because there is a write up to newsletter for customers in there but in case of GDPR subscribers need to double opt-in for be a subscriber of newsletter. I asked that for two years and it was promised as a idea for next releases in future. I still wait for that. But I pay each year $99 for upgrades which are at all only bug fixes. This shouldn't cost me anything, because fixes should be free. I like to pay for new functions in it.

What I like to say here is, if 1. extensions should have better descriptions about what they do (for me it is often unclear and after I bought them I gave them back and requested refund) and 2. extensions devs should integrated ideas from customers in shorter cicles. I would like to see better in quality coded extensions and promoting them at woocommerce.com blog. And please listen to customers.

If you follow my whishes it would gave devs more income. Promoting a product that is not as customers like them to be will not bring more sells.

**Comment by franticpsyx** [comment]
@ajgw -- glad to see you here! We've chatted via the SW help desk a few times in the past 🙂 

> But think why do extensions to be promoted more?

To add another angle to this discussion, I'd like to share an observation (still biased, like all opinions voiced here). Having supported thousands of store owners and developers over the last few years, I find that the biggest hurdle that newcomers to WordPress and WooCommerce must overcome to succeed with the platform is this: 

WordPress makes it too easy to find, install and eventually become dependent on low-quality software. The life of the average WooCommerce store owner is riddled with notices, ads, bloated UIs, poor-performing code, and sometimes security issues. The biggest strengths of Wordpress have always been its biggest weaknesses -- the size and diversity of the plugin ecosystem is one of them.

WooCommece.com extensions aren't perfect -- generic solutions to complex problems can't be perfect. However, the average quality on offer on WooCommerce.com is much more likely to put the average WooCommerce newcomer on a path to success. This opinion is based on some facts:

* Extension submissions on WooCommerce.com are carefully vetted. The process ensures that whatever ends up on WooCommerce.com is fit-for-purpose.
* Reviews are real and can't be gamed with.
* The Ideas boards (and change-logs) are public. Anyone can browse through open & completed ideas, determine how different Vendors manage their products, and make up his/her mind based on facts.

WooCommerce has a business side to it like every freemium plugin on WordPress.org 🙂. If WooCommerce.com is providing a lower-risk, higher-quality path to the **average** user building his/her own WooCommerce store, then everyone wins if this path is made easier to discover.

Of course, if disruptive design is a sign of low-quality software, then choosing disruptive design tactics as a means of promoting high-quality software is, I admit, a bit paradoxical 😄

**Comment by ajgw** [comment]
@franticpsyx  I am with you Manos. I’ve spend the past thousands of Dollars at woocommerce.com. I tried in-between free solutions and solutions from other marketplaces and yes, all the time I came back to woocommerce.com. I manage about twenty sites for clients and few of my own. I see how clients make their first steps into Wordpress and Woocommerce and which questions they ask. And if a solution for their wish will be solved with a extension, I recommend ever woocommerce.com extensions. But how does a new user know what a product bundle will do if they don’t know which products they like to sell and the workflow should be setup. The answer will be: they ask me how to do that. And I will answer them with a extension from woocommerce.com if there is one. We help others to find their way to a working ecommerce solution and we know where to search for good plugins.

On the other side if you promote extensions in lists, how will a newbie user accept the higher prices for extensions from woocommerce.com if storeowner has limited capabilities in english language (most of my clients don’t understand descriptions for your extensions because it is very technical and require some expertise) and if customers find free extensions which seems to do the same.

If at least we – who know the quality of extensions from woocommerce.com and do the job for clients – buy, install and configure them, why do we have to deal with promoting ads?

**Comment by ajgw** [comment]
Beside note to all devs: Has anyone from you all ever translated the wonderful Subscriptions extension to another language? If I am looking for a new extensions anywhere or a theme, my first search is about is it translated to my own language. If not, I leave. Thats a big marketing argument. Or switch woocommerce.com to multilingual and tell customers in their native language about your wonderful extensions so they understand what you wrote. 

Hey, my idea first is create site for submitting language files for extensions.

**Comment by lukecav** [comment]
Not a fan of the in product list extension nag.
![Extension add in product list](https://user-images.githubusercontent.com/5797332/55733962-f1be4800-59db-11e9-8711-63c1403e212c.png)

**Comment by timmyc** [comment]
ICYMI RC2 of 3.6 [was released yesterday](https://woocommerce.wordpress.com/2019/04/10/woocommerce-3-6-rc2/), and includes a number of updates that were a result of feedback in this issue, and from conversations with the plugin team.

Most notably the following items are now in place for 3.6:

- Suggestions are no longer shown on the Product Listing page. #23211
- Display of suggestions can now be turned off permanently by a store setting which is found on a new settings page: Settings > Advanced > WooCommerce.com. Suggestions are still shown by default, but can be turned off via that setting or via the filter.
- Verbiage around the opt in/out of Usage Tracking was also updated.

As such, I'm going to close out this issue. If anyone has further changes they would like seen to this feature, feel free to open up a new issue or pull request. We have already put it on the Dev Chat agenda for this month too so we can chat more about it then.

In closing - thanks again for your feedback - and extra special thanks for those that chimed in on the various pull requests. And if you haven't done so yet, please test RC2 🙌

**Comment by jessuppi** [comment]
A few comments, since I'm arriving late:

1. Is WooCommerce the new Jetpack? Mullenweg fingerprints all over these sly moves...

2. Once again @Ipstenu the only sentient, reasonable voice at WP.org... this is why there needs to be more women (and beyond) involved in WordPress = diversity of thought etc.

3. WooCommerce tracking and [suggestions](https://github.com/woocommerce/woocommerce/pull/23218) are disabled in next Dashboard Cleanup release, which continues to expand focus on WC Core (along with WP Core). The silver lining of bad WC decisions is that at least they always provide hooks/filters to disable the bloat, hopefully this doesn't go the way of WP Core where more and more features are hard-coded ;)

https://github.com/littlebizzy/dashboard-cleanup
